ui/components/multichain/dapp-permission-modal/dapp-permission-modal.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { flatten } from 'lodash';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Display,
+  FlexDirection,
+  BorderColor,
+  TextVariant,
+  BlockSize,
+  AlignItems,
+  JustifyContent,
+} from '../../../helpers/constants/design-system';
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Modal,
+  Box,
+  Text,
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarAccountVariant,
+  ModalFooter,
+  Button,
+  IconName,
+  ButtonVariant,
+} from '../../component-library';
+import {
+  getDappPermissionModal,
+  getPermissionsForActiveTab,
+  getOriginOfCurrentTab,
+  getUseBlockie,
+  getPermissionSubjects,
+} from '../../../selectors';
+import { shortenAddress } from '../../../helpers/utils/util';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  hideDappPermissionModal,
+  removePermittedAccount,
+} from '../../../store/actions';
+import Confusable from '../../ui/confusable';
+import { getPermissionDescription } from '../../../helpers/utils/permission';
+import PermissionCell from '../../app/permission-cell';
+
+export const DappPermissionModal = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+  const permissionSubjects = useSelector(getPermissionSubjects);
+  const permissions = useSelector(getPermissionsForActiveTab);
+  const useBlockie = useSelector(getUseBlockie);
+  const { open, account } = useSelector(getDappPermissionModal);
+
+  function closeModal() {
+    dispatch(hideDappPermissionModal());
+  }
+
+  const permissionLabels = flatten(
+    permissions.map(({ key, value }) =>
+      getPermissionDescription({
+        t,
+        permissionName: key,
+        permissionValue: value,
+        targetSubjectMetadata: permissionSubjects,
+      }),
+    ),
+  );
+
+  return (
+    <Modal isOpen={open} onClose={closeModal}>
+      <ModalOverlay />
+      <ModalContent
+        modalDialogProps={{
+          display: Display.Flex,
+          flexDirection: FlexDirection.Column,
+        }}
+      >
+        <ModalHeader
+          paddingBottom={4}
+          paddingRight={4}
+          paddingLeft={4}
+          onClose={closeModal}
+        >
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Row}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.center}
+          >
+            <AvatarAccount
+              borderColor={BorderColor.transparent}
+              size={AvatarAccountSize.Sm}
+              address={'0x00'}
+              variant={
+                useBlockie
+                  ? AvatarAccountVariant.Blockies
+                  : AvatarAccountVariant.Jazzicon
+              }
+              marginInlineEnd={2}
+            />
+            <Text variant={TextVariant.headingSm}>
+              {account.label ? (
```

Comment:
This is not rendering account name, instead just showing the address. 

![Uploading Screenshot 2024-03-12 at 12.57.13 PM.jpg…]()